clang/lib/AST/Mangle.cpp

```diff
@@ -29,6 +29,24 @@
 
 using namespace clang;
 
+void clang::mangleObjCMethodName(raw_ostream &OS, bool includePrefixByte,
+                                 bool isInstanceMethod, StringRef ClassName,
+                                 std::optional<StringRef> CategoryName,
+                                 StringRef MethodName, bool isThunk) {
+  // \01+[ContainerName(CategoryName) SelectorName]
+  if (includePrefixByte)
+    OS << "\01";
+  OS << (isInstanceMethod ? '-' : '+');
+  OS << '[';
+  OS << ClassName;
+  if (CategoryName)
+    OS << "(" << *CategoryName << ")";
+  OS << " ";
+  OS << MethodName;
+  OS << ']';
+  if (!isThunk)
+    OS << "_inner";
```

Comment:
I'm confused about this. You're adding `_inner` to everything that *isn't* a thunk? Now, obviously, from the lack of sweeping changes to the entire ObjC test suite, that's not actually what's happening, but I'd guess it's because you're only calling this for direct methods, which is not at all clear from the method name.

`inner` is a really non-specific thing to add to the symbol name. This is important because it's what's going to show up in backtraces. Maybe `_nonnull` would be better?

Although... the end state we want is that the non-thunk function gets the undecorated name and the thunk function gets the decorated one, right?

Reply:
> end state we want is that the non-thunk function gets the undecorated name and the thunk function gets the decorated one, right?

I think it's the other way around, thunk functions gets undecorated ones so anyone can use the old function name when they don't know if the receiver will be nil.